### ISSUE TITLE
Why classification loss is divided by num_boxes instead of num_query

### ISSUE THREAD

**Issue by Ryanshuai** [lyuwenyu/RT-DETR#388]
https://github.com/lyuwenyu/RT-DETR/blob/9f107dab37c402962a4632570e63ec83c30e1ecc/rtdetr_pytorch/src/zoo/rtdetr/rtdetr_criterion.py#L135

在rtdetr_criterion.py的第135行,分类损失被除以num_boxes而不是num_queries:
loss = loss.mean(1).sum() * src_logits.shape[1] / num_boxes

想问下作者，没找到其他code对应的这部分，所以找不到出处原因； 看 DETR 的分类loss，貌似是在query上取平均？
如题，在分类loss 里，为什么要 /num_boxes 而不是 /num_query？
1.  分类应该是对所有的query做分类，那为什么不是在query上平均？
2. /num_boxes 这种设计选择的原因是什么?它与 /num_query 相比有什么优势?
3. 使用num_boxes而不是固定的num_queries是否会引入训练不稳定性,特别是当不同图像中的目标数量差异很大时?
4. 是否有任何消融实验或比较来支持这个设计选择?

看回复时间线，感觉作者很忙，感谢作者解惑！

**Comment by lyuwenyu** [comment]
没有做过相关消融；

1. detr采用的是ce 每个query都有对应的类别 （包括背景类）；所以除num_query
2. 我们这里loss是采用的focal-loss 并不包含背景类； 所以除正样本数（num_boxes）
